src/main/java/org/elasticsearch/common/util/concurrent/PrioritizedEsThreadPoolExecutor.java

```diff
@@ -161,6 +168,39 @@ public int compareTo(PrioritizedRunnable pr) {
             }
             return insertionOrder < ((TieBreakingPrioritizedRunnable) pr).insertionOrder ? -1 : 1;
         }
+
+        public void scheduleTimeout(ScheduledExecutorService timer, final Runnable timeoutCallback, TimeValue timeValue) {
+            this.timeoutCallback = timeoutCallback;
+            synchronized (this) {
+                timeoutFuture = timer.schedule(new Runnable() {
```

Comment:
Currently, that's not possible since we call scheduleTimout first and only upon its return we execute, but we can add the check for safety in case somebody will flip the order of things in future revisions.